spring-integration-core/src/test/java/org/springframework/integration/support/management/graph/IntegrationGraphServerTests.java

```diff
@@ -165,6 +171,41 @@ public PollerMetadata defaultPoller() {
 			return poller;
 		}
 
+		@Bean
+		@Router(inputChannel = "four")
+		public HeaderValueRouter router() {
+			HeaderValueRouter router = new HeaderValueRouter("foo");
+			router.setChannelMapping("bar", "barChannel");
+			router.setChannelMapping("baz", "bazChannel");
+			router.setDefaultOutputChannel(discards());
+			return router;
+		}
+
+		@Bean
+		@Router(inputChannel = "four")
+		public RecipientListRouter rlRouter() {
+			RecipientListRouter router = new RecipientListRouter();
+			router.setRecipients(
+					Arrays.asList(new Recipient[] { new Recipient(barChannel()), new Recipient(bazChannel()) }));
```

Comment:
Yeah... That's why in Java DSL I have `DslRecipientListRouter` variant with some bunch of `add()` methods and late channel resolution logic.
We can absorb it in SI-5.0